### PR TITLE
Pytorch 2.3 upgrade

### PR DESCRIPTION
**Pull request by suzhoum**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |
| botocore==1.34.114 | botocore==1.34.115 |
| boto3==1.34.114 | boto3==1.34.115 |
| sympy==1.12 | sympy==1.12.1 |
| botocore==1.34.114 | botocore==1.34.115 |
| boto3==1.34.114 | boto3==1.34.115 |
| sympy==1.12 | sympy==1.12.1 |

**Comment by suzhoum** [comment]
/benchmark module=multimodal preset=stable benchmark=none time_limit=g4_12x

**Comment by suzhoum** [comment]
/benchmark module=multimodal preset=multimodal_best benchmark=tabular_full time_limit=4h

**Comment by suzhoum** [comment]
/benchmark module=tabular preset=tabular_best benchmark=tabular_full time_limit=1h

**Comment by prateekdesai04** [comment]
/benchmark module=tabular preset=tabular_best benchmark=tabular_full time_limit=1h

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |
| torchvision==0.18.0 | torchvision==0.17.2 |
| lightning==2.2.5 | lightning==2.1.4 |
| triton==2.3.0 | triton==2.2.0 |
| torch==2.3.0 | torch==2.2.2 |
| nvidia-nccl-cu12==2.20.5 | nvidia-nccl-cu12==2.19.3 |
| torchvision==0.18.0 | torchvision==0.17.2 |
| lightning==2.2.5 | lightning==2.1.4 |
| triton==2.3.0 | triton==2.2.0 |
| torch==2.3.0 | torch==2.2.2 |
| nvidia-nccl-cu12==2.20.5 | nvidia-nccl-cu12==2.19.3 |

**Comment by shchur** [comment]
@suzhoum FYI I have rebased the PR branch on `master` to include our latest commits to time series

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |
| torchvision==0.18.1 | torchvision==0.17.2 |
| lightning==2.2.5 | lightning==2.1.4 |
| triton==2.3.1 | triton==2.2.0 |
| torch==2.3.1 | torch==2.2.2 |
| nvidia-nccl-cu12==2.20.5 | nvidia-nccl-cu12==2.19.3 |
| torchvision==0.18.1 | torchvision==0.17.2 |
| lightning==2.2.5 | lightning==2.1.4 |
| triton==2.3.1 | triton==2.2.0 |
| torch==2.3.1 | torch==2.2.2 |
| nvidia-nccl-cu12==2.20.5 | nvidia-nccl-cu12==2.19.3 |

**Comment by shchur** [comment]
It seems that `gluonts~=0.15.1` currently caps `pytorch_lightning<2.2`
https://github.com/awslabs/gluonts/blob/dev/requirements/requirements-pytorch.txt#L4

We will need to raise this cap and re-release GluonTS to enable PyTorch 2.3 support on our side.

**Comment by shchur** [comment]
We probably need a stricter lower bound on `lightning` based on this CVE https://github.com/awslabs/gluonts/security/dependabot/23

**Comment by suzhoum** [comment]
> We probably need a stricter lower bound on `lightning` based on this CVE https://github.com/awslabs/gluonts/security/dependabot/23

Thanks, I just updated the lower bound.

**Review by prateekdesai04 — APPROVED** [review]
LGTM!

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |
| botocore==1.34.121 | botocore==1.34.122 |
| boto3==1.34.121 | boto3==1.34.122 |
| google-auth==2.29.0 | google-auth==2.30.0 |
| tensorboard==2.16.2 | tensorboard==2.17.0 |
| typing_extensions==4.12.1 | typing_extensions==4.12.2 |
| botocore==1.34.121 | botocore==1.34.122 |
| boto3==1.34.121 | boto3==1.34.122 |
| google-auth==2.29.0 | google-auth==2.30.0 |
| tensorboard==2.16.2 | tensorboard==2.17.0 |
| typing_extensions==4.12.1 | typing_extensions==4.12.2 |

File: core/src/autogluon/core/_setup_utils.py [hunk 1]
```diff
@@ -26,9 +26,9 @@
     "networkx": ">=3.0,<4",  # Major version cap
     "tqdm": ">=4.38,<5",  # Major version cap
     "Pillow": ">=10.0.1,<11",  # Major version cap
-    "torch": ">=2.2,<2.3",  # Major version cap, sync with common/src/autogluon/common/utils/try_import.py
-    "lightning": ">=2.1,<2.2",  # Major version cap
-    "pytorch_lightning": ">=2.2,<2.3",  # Major version cap, capping `lightning` does not cap `pytorch_lightning`!
+    "torch": ">=2.3,<2.4",  # Major version cap, sync with common/src/autogluon/common/utils/try_import.py
```
Comment (Innixma) [inline core/src/autogluon/core/_setup_utils.py:29]: Why are we pinning torch to >=2.3? This seems much too strict.

**Review by Innixma — CHANGES_REQUESTED** [review]

File: common/src/autogluon/common/utils/try_import.py [hunk 2]
```diff
@@ -130,7 +130,7 @@ def try_import_torch():
         raise ImportError(
             "Unable to import dependency torch\n"
             "A quick tip is to install via `pip install torch`.\n"
-            "The minimum torch version is currently 2.1."  # sync with core/_setup_utils.py
+            "The minimum torch version is currently 2.3."  # sync with core/_setup_utils.py
```
Comment (Innixma) [inline common/src/autogluon/common/utils/try_import.py:133]: Why are we pinning torch to >=2.3? This seems much too strict.

File: core/src/autogluon/core/_setup_utils.py (hunk 1 above)
Comment (suzhoum) [reply core/src/autogluon/core/_setup_utils.py:29]: I checked the previous releases and we have been restricting `torch` to be within one major version. If you feel this is too strict, i can make ti to be >=2.2

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |
| torchvision==0.18.1 | torchvision==0.17.2 |
| lightning==2.2.5 | lightning==2.1.4 |
| triton==2.3.1 | triton==2.2.0 |
| torch==2.3.1 | torch==2.2.2 |
| nvidia-nccl-cu12==2.20.5 | nvidia-nccl-cu12==2.19.3 |
| torchvision==0.18.1 | torchvision==0.17.2 |
| lightning==2.2.5 | lightning==2.1.4 |
| triton==2.3.1 | triton==2.2.0 |
| torch==2.3.1 | torch==2.2.2 |
| nvidia-nccl-cu12==2.20.5 | nvidia-nccl-cu12==2.19.3 |

**Review by Innixma — APPROVED** [review]
LGTM

**Pull request merged**